shared/src/main/java/model/Game.java

```diff
@@ -1,71 +1,94 @@
 package model;
 
+import clientProxy.IClientIsInGame;
+
+import java.util.Set;
+import java.util.TreeSet;
 import java.util.UUID;
 
 public class Game {
     private UUID gameId;
     private boolean isPlaying;
     private Player host;
-    private Player[] playerList;
+    private Set<Player> playerList = new TreeSet<>();
     private int maxPlayers;
     private int currentPlayers;
     private String gameName;
 
-    public Game(Player host, int maxPlayers, int currentPlayers, String gameName) {
+    public Game(Player host, String gameName) {
         this.host = host;
-        this.maxPlayers = maxPlayers;
-        this.currentPlayers = currentPlayers;
         this.gameName = gameName;
+        this.isPlaying = false;
+        this.maxPlayers = 5;
+        this.currentPlayers = 0;
+        this.gameId = UUID.randomUUID();
+
+        addPlayer(host);
     }
 
-    public UUID getGameId() {
-        return gameId;
+    IClientIsInGame clientProxy = new IClientIsInGame() {
```

Comment:
Just so it's clear, this will point to an instance called `ClientProxy` and not an anonymous class as you have here.